File: 2 Arrays/Cyclic Rotation/CyclicRotation.java
class Solution {
  public int[] solution(int[] A, int K) {
    int[] result = new int[A.length];

    for (int i = 0; i < result.length; i++) {
      int newPosition = (i + K) % result.length;
      result[newPosition] = A[i];
    }
  
    return result;
  }
}

File: 2 Arrays/Cyclic Rotation/CyclicRotation.js
function solution(A, K) {
  if (!A.length) return [];
  for (let i = 0; i < K; i++) {
    const tail = A[A.length - 1];
    console.log(A[A.length - 1]);
    console.log('tail', tail);
    A.pop();
    A.unshift(tail);
  }
  return A;
}

console.log(solution([1, 2, 3, 4, 5, 6, 7, 8, 9, 10], 3));
console.log(solution([3, 8, 9, 7, 6], 3));
